reflred/footprint.py

```diff
@@ -287,17 +287,20 @@ def _trapezoid_sum_right(p, w1, w2):
     """
     Find the integral from p to inf of the trapezoid defined by w1 and w2.
     """
-    slope = 1.0/(w1-w2)
+    slope = 1.0/(w1-w2) if not w1 == w2 else None
+    SUM = 0
     a = np.maximum(p, -w1) + w1
     b = np.maximum(p, -w2) + w1
-    A = 0.5*slope*(b**2-a**2)*(b > a)
+    if slope is not None:
+        SUM += 0.5*slope*(b**2-a**2)*(b > a)
     a = np.maximum(p, -w2) + w2
     b = np.maximum(p, w2) + w2
-    A += (b-a)*(b > a)
+    SUM += (b-a)*(b > a)
     a = w1 - np.maximum(p, w1)
     b = w1 - np.maximum(p, w2)
-    A += 0.5*slope*(b**2-a**2)*(b > a)
-    return A
+    if slope is not None:
+        SUM += 0.5*slope*(b**2-a**2)*(b > a)
+    return SUM
```

Comment:
I think this is less confusing done as a single condition:
```python
assert w1 >= w2 >= 0, "trapezoid [-w1, -w2, w2, w1] requires w1 >= w2 >= 0"
# Integral bounds. Note that clipping the intervals to lie within the bounds may
# yield a pair of end points outside the interval range but the formulas for the
# area of each interval will be zero if that is the case.
bounds = (p, np.inf)
# rising triangle over [-w1, -w2] shifted to [0, w1-w2]  
ra, rb = w1 + np.clip([-w1, -w2], *bounds)
# flat midsection over [-w2, w2]
ma, mb = np.clip([-w2, w2], *bounds)
# falling triangle over [w2, w1] reversed to [0, w1-w2]
fa, fb = w1 - np.clip([w1, w2], *bounds)
if abs(w1-w2) < 1e-10:  # approximately square
    return mb - ma
else: # trapezoidal
    slope = 1/(w1 - w2)  # slope of the wings
    return 0.5*slope*(rb-ra)*(rb+ra) + (mb - ma) + 0.5*slope*(fb-fa)*(fb+fa)
```

Reply:
Yes, the embedded documentation makes it much more clear.  It took me a while to figure out what that function was doing, absent the comments.

Reply:
Looking a couple of lines above to how it is called I think we can further modify the function to take the bounds `[p_far, p_near]` rather than calling `trapsum(p_near) - trapsum(p_far)`. Looking further up I see that `p_near ≤ p_far`, so I'm guessing the original code is wrong (or if not wrong, very misleading). The `_apply_footprint` function is ignoring negative footprint values, so it will silently do nothing.